src/snowcli/cli/nativeapp/policy.py

```diff
@@ -0,0 +1,31 @@
+from dataclasses import dataclass
+
+import typer
+
+
+@dataclass
+class TaskContext:
+    user_prompt: str
+    is_interactive_mode: bool
+
+
+class PolicyBase:
+    exit_code: int = 0
+
+    def should_proceed(self, ctx: TaskContext):
+        pass
+
+
+class AllowAlwaysPolicy(PolicyBase):
+    def should_proceed(self, ctx: TaskContext):
+        return True
+
+
+class AlwaysAskPolicy(PolicyBase):
+    def should_proceed(self, ctx: TaskContext):
+        if ctx.is_interactive_mode:
+            should_continue = typer.confirm(ctx.user_prompt)
+            return should_continue
+        else:
+            self.exit_code = 1
```

Comment:
Other possible approaches:
1. Use a tuple return type (T/F, exit_code)
2. Use enum values as return type.
The above two are in case we want no side-effects, such as re-assigning `exit_code` to 1.